packages/vendure-order-client/src/vendure-order-client.ts

```diff
@@ -364,9 +364,36 @@ export class VendureOrderClient<A = unknown> {
       this.queries.GET_ORDER_BY_CODE,
       { code }
     );
+    if (
+      orderByCode?.state !== 'PaymentSettled' &&
+      orderByCode?.state !== 'PaymentAuthorized'
+    ) {
+      throw Error(`Can not get order by code.`);
+    }
     return orderByCode;
   }
 
+  /**
+   * Poll order by code for a given poll amount of times. And Poll for x seconds
+   */
+  @HandleLoadingState('$activeOrder')
+  async pollOrderByCode(
+    code: string,
+    pollFor: number = 10
+  ): Promise<ActiveOrder<A>> {
+    let order: ActiveOrder<A> | undefined;
+    let pollingCount = 0;
+    while (!order) {
+      if (pollingCount > pollFor) {
+        throw Error(`Order not settled after polling ${pollFor} times`);
+      }
+      order = await this.getOrderByCode(code);
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      pollingCount++;
+    }
+    return order;
```

Comment:
Maybe it's a good idea to accept an extra argument:

```ts
async pollOrderByCode(
    code: string,
    pollFor: number = 10,
    acceptedStates: string[] = ['PaymentSettled', 'PaymentAuthorized']
  )
```

This way we can also pass in `AwaitingPrepress` as extra state in the CG project, because thats also a valid state, but only in the CG project.

We would then use the states in this function: 

```ts
if ( !acceptedStates.includes(order.state) {
   // poll again
}